### PR TITLE
Feature/product routes

### PR DESCRIPTION
**Pull request by DanaXCohen**

### CODE REVIEW

File: server/controllers/productController.js [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
-    const userId = req.user.userId;
-
-    // Validating inputs here...
+export async function createProduct(req, res, next) {
+    const { name, category, status, description, price, userId } = req.body;
 
     try {
-        const user = await User.findById(userId);
 
+        if (!mongoose.Types.ObjectId.isValid(userId)) {
```
Comment (adiritzhakii) [inline server/controllers/productController.js:12]: Did you check it ?

File: server/controllers/productController.js [hunk 2]
```diff
@@ -27,3 +35,91 @@ export async function createProduct(req, res)  {
...
+        }
+
+        res.status(200).json({ product });
+    }
+    catch (error) {
+        res.status(500).json({ msg: 'Server error' });
+    }
+}
+
+export async function getAlProducts(req, res) {
```
Comment (adiritzhakii) [inline server/controllers/productController.js:118]: Fix name

File: server/models/user.js [hunk 3]
```diff
@@ -22,6 +23,9 @@ userSchema.pre('save', async function(next) {
     try {
         const salt = await bcrypt.genSalt(10);
```
Comment (adiritzhakii) [inline server/models/user.js:24]: The salt should save in db... how you gonna decrypt it without salt.

**Review by adiritzhakii — CHANGES_REQUESTED** [review]

File: server/controllers/productController.js (hunk 1 above)
Comment (DanaXCohen) [reply server/controllers/productController.js:12]: yes

File: server/models/user.js (hunk 3 above)
Comment (DanaXCohen) [reply server/models/user.js:24]: what do you mean?

**Pull request merged**